actions/shared/verify_installation/action.yml

```diff
@@ -0,0 +1,162 @@
+              break
+            fi
+            sleep ${{inputs.timeout}}
+          done
+          if [[ "$status" != 'Succeeded' ]]; then
+            echo $provisioner_pod pod has not Succeeded status
+          fi
+          echo PROVISIONER LOGS:
+          kubectl logs $provisioner_pod -n ${{inputs.namespace}}
+          logs=$(kubectl logs $provisioner_pod -n ${{inputs.namespace}})
+          if [[ $logs != *"Failed components statuses are []"* ]]; then
+            echo Not all components are in succeed status after ${{inputs.max_attempts}} attempts
+            echo PODS:
+            kubectl get pods -n ${{inputs.namespace}}
+            exit 1
+          fi
+        else
+          echo provisioner pod is not found
+        fi
+      shell: bash
+
+    - name: Get logs from all containers in pods
+      run: |
+        #GET LOGS FROM ALL CONTAINERS IN PODS
+        file_name=artifacts/${{inputs.namespace}}_log_from_containers.txt
+        IFS=' ' read -ra pods <<< $(kubectl get pods --no-headers -o custom-columns=":metadata.name" -n ${{inputs.namespace}} | tr $'\n' ' ')
+        echo "Logs from all containers in pods: " > $file_name
+        for pod in "${pods[@]}"; do
+            IFS=' ' read -ra containers <<< $(kubectl get pod "$pod" -n ${{inputs.namespace}} -o jsonpath="{.spec.containers[*].name}")
+            for container in "${containers[@]}"; do
+              echo $pod $container >> $file_name
+              kubectl logs "$pod" -n ${{inputs.namespace}} -c "$container" >> $file_name
+            done
+        done
+      shell: bash
+
+    - name: Get logs from test pod
+      run: |
+        #GET LOGS FROM TEST POD
+        file_name=artifacts/${{inputs.namespace}}_tests.txt
+        IFS=' ' read -ra pods <<< $(kubectl get pods --no-headers -o custom-columns=":metadata.name" -n ${{inputs.namespace}} | tr $'\n' ' ')
+        for pod in "${pods[@]}"; do
+          if [[ $pod == *"tests"* ]]; then
+            test_pod=$pod
+          fi
+        done
+        if [[ $test_pod ]]; then
+          echo "test_pod=$test_pod"
+          for i in {1..${{inputs.max_attempts}}}; do
+            status=$(kubectl get pod $test_pod -n ${{inputs.namespace}} -o jsonpath="{.status.phase}")
+            if [[ "$status" == 'Running' ]]; then
+              break
+            fi
+            sleep ${{inputs.timeout}}
+          done
+          for i in {1..${{inputs.max_attempts}}}; do
+            logs=$(kubectl logs $test_pod -n ${{inputs.namespace}})
+            if [[ $logs == *"Report: /opt/robot/output/report.html"* ]]; then
+              kubectl logs $test_pod -n ${{inputs.namespace}} > $file_name
+              break
+            fi
+            sleep ${{inputs.timeout}}
+          done
+          echo 'TEST LOGS: '
+          kubectl logs $test_pod -n ${{inputs.namespace}}
+          if [[ $logs == *"| FAIL |"* ]]; then
+            exit 1
+          fi
+        else
+          echo There is no test pod
+        fi
```

Comment:
Same here: Have you considered `if ! [[ $test_pod ]]; then echo test_pod is empty; exit 1`?

Reply:
We can install service without tests, 'exit 1' is not correct